### PR TITLE
Config#15/create a project folder structure

### LINKED ISSUE #15: Create a project folder structure(monerepo)
Stack
- NestJs
- NextJs
- MongoDB
- Typeorm
- TailwindCSS

### PR DESCRIPTION
**Pull request by pierratono**
## 🚀 Pull Request

**Changes Made:**
In this pull request, I have configured the monorepo structure for NestJS with Swagger for API documentation and NextJS with Tailwind CSS for styling.

**Why this PR?**
This pull request aims to enhance the project structure by adopting a monorepo approach, making it more modular and maintainable. The integration of Swagger facilitates better API documentation, improving the overall developer experience.

**Which issue does it resolve?**
Resolves #15 

**Dev Notes:**

To run you have to use `pnpm dev` inside root project
To install a new dependency use  pnpm [filtering](https://pnpm.io/filtering)`--filter=api` or `--filter=web`

**Checklist:**

- [x] Tested locally
- [x] Follows coding standards
- [ ] Documentation updated

### CODE REVIEW

**Review by Musigwa — CHANGES_REQUESTED** [review]
Hey @pierratono thanks for submitting the PR. A few points of concern here:

Your branch's git history is not well aligned with the previous work history as shown in the below screenshot

    a. The brown arrow shows your actual 'commit'
    b. The blue and green arrows show commits that came from the recently closed PRs which shouldn't be the case since they were merged into the base branch "main"
    c. The pink arrow shows your most recent commit again
    
Screenshot 2024-02-02 at 10 49 00 AM

Action items:
1. Please consider resetting your branch’s git history and rewind it to the base branch correctly

2. I noticed that the folder structure you suggested can be tweaked a little to make even more sense. What do you think of the below folder structure adjustment?

```bash
wishop/
|-- packages/
|   |-- api/
|   |   |-- __tests__/
|   |   |-- controllers/
|   |   |-- modules/
|   |   |-- services/
|   |   |-- tsconfig.json
|   |   |-- package.json
|
|   |-- web/
|   |   |-- __tests__/
|   |   |-- pages/
|   |   |-- components/
|   |   |-- styles/
|   |   |-- public/
|   |   |-- next.config.js
|   |   |-- package.json
|
|-- shared/
|   |-- models/
|   |-- utilities/
|   |-- ...
|
|-- tsconfig.base.json
|-- package.json
|-- nest-cli.json
|-- .gitignore
|-- .prettierrc
|-- .eslintrc
```

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+node_modules
+dist
+build
+.env
+.env.local
+.env.production
+.env.testing
+.env.staging
+pnpm-lock.yaml
+public/*
```
Comment (Nziranziza) [inline .gitignore:10]: Add new line at the end of file

File: .hintrc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": [
+    "development"
+  ],
+  "hints": {
+    "typescript-config/strict": "off",
+    "typescript-config/is-valid": "off"
+  }
+}
```
Comment (Nziranziza) [inline .hintrc:9]: Add new line at the end of file

File: apps/api/.env.example [hunk 3]
```diff
@@ -0,0 +1 @@
+MONGODB_URI='mongodb://localhost/wishop'
```
Comment (Nziranziza) [inline apps/api/.env.example:1]: Add new line at the and of file

File: apps/api/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+public/*
```
Comment (Nziranziza) [inline apps/api/.gitignore:1]: Add new line at the end of file

File: apps/api/.prettierrc [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
+  "trailingComma": "all"
+}
```
Comment (Nziranziza) [inline apps/api/.prettierrc:4]: Add new line at the end of file

**Review by Nziranziza — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Musigwa) [reply .gitignore:10]: Good catch @Nziranziza

**Pull request closed without merging**